draft-ietf-oauth-v2-1.md

```diff
@@ -2283,56 +2305,24 @@ audience), typically a single resource server (or a list of resource
-client MUST validate the TLS certificate chain when making requests
-to protected resources, including checking the Certificate Revocation
-List (CRL) {{RFC5280}}.
-
-If cookies are transmitted without TLS protection, any
-information contained in them is at risk of disclosure.  Therefore,
-Bearer tokens MUST NOT be stored in cookies that can be sent in the
-clear, as any information in them is at risk of disclosure.
-See "HTTP State Management Mechanism" {{RFC6265}} for security
-considerations about cookies.
-
-In some deployments, including those utilizing load balancers, the
-TLS connection to the resource server terminates prior to the actual
-server that provides the resource.  This could leave the token
-unprotected between the front-end server where the TLS connection
-terminates and the back-end server that provides the resource.  In
-such deployments, sufficient measures MUST be employed to ensure
-confidentiality of the access token between the front-end and back-end
-servers; encryption of the token is one such possible measure.
-
-To deal with access token capture and replay, the following recommendations
-are made: First, the lifetime of the token MUST be limited; one means
-of achieving this is by putting a validity time field inside the
-protected part of the token.  Note that using short-lived 
-tokens reduces the impact of them being leaked.  Second,
-confidentiality protection of the exchanges between the client and
-the authorization server and between the client and the resource
-server MUST be applied.  As a consequence, no eavesdropper along the
-communication path is able to observe the token exchange.
-Consequently, such an on-path adversary cannot replay the token.
-Furthermore, when presenting the token to a resource server, the
-client MUST verify the identity of that resource server, as per {{BCP195}}
-and Section 3.1 of "HTTP Over TLS" {{RFC2818}}.  Note that the client MUST
-validate the TLS certificate chain when making these requests to
-protected resources.  Presenting the token to an unauthenticated and
-unauthorized resource server or failing to validate the certificate
-chain will allow adversaries to steal the token and gain unauthorized
-access to protected resources.
+{{communication-security}} provides information to
+protect against access token disclosure and providing
+confidentiality and integrity for the communications
+between client, resource server and authorization server.
+
+The following RECOMMENDATIONS mitigate the risk of
+access token theft and replay:
+
+1. limit the lifetime of the token, e.g. by putting a validity time field inside the
+   protected part of the token. Short-lived tokens reduces both the probablity and the impacts of a leak.
+2. protect the confidentiality of the communication between the client and the RS
+   to mitigate the risk of eavesdropping;
+3. the client MUST verify the identity of the resource server before
```

Comment:
To this end, one thing which is currently lacking is a standard/recommended way for the AS to communicate _to the client_ the intended audience(s) (URLs) of the issued access token.

Reply:
@randomstuff that is what resource indicators and RAR do. 
Audience restriction is always a good idea but it will only limit the impact of replay, it does not prevent it. Most application with on substantial security requirements will use sender constrained access tokens.

Reply:
@tlodderstedt, in the current resource indicators RFC, the client can tell the AS the intended audience (RS) but there is currently no standard way for the *AS* to communicate the audience of the token to the *client* (for example as a confirmation that the audience restriction has been applied by the AS).